packages/orchestrator/internal/sandbox/uffd/userfaultfd/userfaultfd_test.go

```diff
@@ -0,0 +1,141 @@
+package userfaultfd
+
+import (
+	"bytes"
+	"fmt"
+	"reflect"
+	"syscall"
+	"testing"
+
+	"go.uber.org/zap"
+
+	"github.com/e2b-dev/infra/packages/orchestrator/internal/sandbox/uffd/fdexit"
+	"github.com/e2b-dev/infra/packages/orchestrator/internal/sandbox/uffd/testutils"
+	"github.com/e2b-dev/infra/packages/shared/pkg/storage/header"
+)
+
+type pageTest struct {
+	pagesize        uint64
+	numberOfPages   uint64
+	operationOffset uint64
+}
+
+func TestUffdMissing(t *testing.T) {
+	tests := []pageTest{
+		// Standard 4K page, operation at start
+		{
+			pagesize:        header.PageSize,
+			numberOfPages:   32,
+			operationOffset: 0,
+		},
+		// Standard 4K page, operation at middle
+		{
+			pagesize:        header.PageSize,
+			numberOfPages:   32,
+			operationOffset: 16 * header.PageSize,
+		},
+		// Standard 4K page, operation at last page
+		{
+			pagesize:        header.PageSize,
+			numberOfPages:   32,
+			operationOffset: 31 * header.PageSize,
+		},
+		// Hugepage, operation at start
+		{
+			pagesize:        header.HugepageSize,
+			numberOfPages:   8,
+			operationOffset: 0,
+		},
+		// Hugepage, operation at middle
+		{
+			pagesize:        header.HugepageSize,
+			numberOfPages:   8,
+			operationOffset: 4 * header.HugepageSize,
+		},
+		// Hugepage, operation at last page
+		{
+			pagesize:        header.HugepageSize,
+			numberOfPages:   8,
+			operationOffset: 7 * header.HugepageSize,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(fmt.Sprintf("pagesize-%d-offset-%d", tt.pagesize, tt.operationOffset), func(t *testing.T) {
+			data, size := testutils.RandomPages(tt.pagesize, tt.numberOfPages)
+
+			uffd, err := newUserfaultfd(syscall.O_CLOEXEC | syscall.O_NONBLOCK)
+			if err != nil {
+				t.Fatal("failed to create userfaultfd", err)
+			}
```

Comment:
Instead of the conditionals, you can use `require.NoError(t, err)` to simplify the tests.